runtime/compiler/optimizer/VectorAPIExpansion.cpp

```diff
@@ -1351,6 +1338,84 @@ TR::Node *TR_VectorAPIExpansion::binaryIntrinsicHandler(TR_VectorAPIExpansion *o
    }
 
 
+
+TR::Node *TR_VectorAPIExpansion::broadcastCoercedIntrinsicHandler(TR_VectorAPIExpansion *opt, TR::TreeTop *treeTop, TR::Node *node,
+                                                        TR::DataType elementType, vec_sz_t vectorLength, handlerMode mode)
+   {
+   TR::Compilation *comp = opt->comp();
+
+   if (mode == checkScalarization)
+      return node;
+
+   if (mode == checkVectorization)
+      {
+      if (!comp->cg()->getSupportsOpCodeForAutoSIMD(TR::vsplats, elementType))
+         return NULL;
+      else
+         return node;
+      }
+
+   if (opt->_trace)
+      traceMsg(comp, "broadcastCoercedIntrinsicHandler for node %p\n", node);
+
+   int32_t elementSize = OMR::DataType::getSize(elementType);
+   TR::Node *valueToBroadcast = node->getChild(3);
+
+   anchorOldChildren(opt, treeTop, node);
+
+   TR::Node *newNode;
+
+   switch (elementType) {
+      case TR::Float:
+          newNode = TR::Node::create(node, TR::ibits2f, 1, TR::Node::create(node, TR::l2i, 1, valueToBroadcast));
+          break;
+      case TR::Double:
+          newNode = TR::Node::create(node, TR::lbits2d, 1, valueToBroadcast);
+          break;
+      case TR::Int8:
+          newNode = TR::Node::create(node, TR::l2b, 1, valueToBroadcast);
+          break;
+      case TR::Int16:
+          newNode = TR::Node::create(node, TR::l2s, 1, valueToBroadcast);
+          break;
+      case TR::Int32:
+          newNode = TR::Node::create(node, TR::l2i, 1, valueToBroadcast);
+          break;
+      case TR::Int64:
+          // redundant conversion to simplify node recreation
+          newNode = TR::Node::create(node, TR::dbits2l, 1, TR::Node::create(node, TR::lbits2d, 1, valueToBroadcast));
+          break;
+      default:
+         TR_ASSERT_FATAL(false, "Was a new element type added to VectorAPI?\n");
```

Comment:
Rather than ask a question, just say the element type was unexpected and ideally reporting the type received.

Reply:
Changed the message.